### ISSUE TITLE
gsconnect / gjs / daemon.js listens only on IPv6 address

### ISSUE THREAD

**Issue by kowalcj0** [andyholmes/gnome-shell-extension-gsconnect#69]
Hi,

Dunno why gsconnect is listening only on the IPv6 address?
Because of this my phone can't find my laptop.

I've opened the ports:
```bash
sudo firewall-cmd --zone=public --permanent --add-port=1716-1764/tcp
sudo firewall-cmd --zone=public --permanent --add-port=1716-1764/udp
```

but as you can see `gjs` is listening only on IPv6 address:
```bash
$ sudo netstat -tnap
Active Internet connections (servers and established)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
...
tcp        0      0 0.0.0.0:111             0.0.0.0:*               LISTEN      1/systemd
tcp        0      0 0.0.0.0:53              0.0.0.0:*               LISTEN      442/dnsmasq         
tcp        0      0 0.0.0.0:41653           0.0.0.0:*               LISTEN      29619/rpc.statd     
tcp        0      0 127.0.0.1:631           0.0.0.0:*               LISTEN      3637/cupsd          
...    
tcp6       0      0 :::1717                 :::*                    LISTEN      23421/gjs 
```

ps. I had no problems with pairing my phone via regular KDE Connect/MConnect.

**Comment by andyholmes** [comment]
GSConnect doesn't actually specify either IPv4 or IPv6 and can accept connections over either. It does this by using `Gio.SocketService.add_inet_port()`:

> Helper function for g_socket_listener_add_address() that creates a TCP/IP socket listening on IPv4 and IPv6 (if supported) on the specified port on all interfaces.

Your output from netstat shows that GSConnect is listening on port 1717:

    tcp6       0      0 :::1717                 :::*                    LISTEN      23421/gjs

Because it's listening on that port instead of the protocol default (1716), the Android app will have difficulty find your phone even though other ports are supported. Probably this means you are also running another KDE Connect server, either `kdeconnectd` or `mconnectd`. You should still be able to connect your phone, but you'll have to manually enter the IP address for your laptop in the Android App:

![manip](https://user-images.githubusercontent.com/1265208/37671023-009fca7a-2c28-11e8-9d14-3fe9f1831982.png)

**Comment by kowalcj0** [comment]
I tried that, but it didn't work.
I also tried connecting after I disabled `ipv6` but it didn't work as well.
I think I messed up with my network config as `kdeconnectd` doesn't work as well.
Will leave this for when I have more spare time :)

**Comment by decaby7e** [comment]
Just wanted to say that this was my exact issue with connecting through a Wireguard VPN. I couldn't figure out why I wasn't getting a connection but it turned out that GSConnect on the desktop wasn't listening on IPv4 and therefore the VPN wouldn't facilitate the connection. Disabling IPv6 fixed the issue. Thank you!

**Comment by guillaume-uH57J9** [comment]
Same issue here with GS Connect v62 and KDE Connect 1.33.2.

GSConnect had trouble connecting for a long time, finally got around to investigating it.
I have firewall rules to only allow specific devices over IPv4, and GJS/GSConnect only listen on IPv6. 

> $ netstat -lpt
> ...
> tcp6       0      0 [::]:1716               [::]:*                  LISTEN      7273/gjs            

Would it be possible to reopen this? Or should I created a new issue?

**Comment by ferdnyc** [comment]
@guillaume-uH57J9 

> Same issue here with GS Connect v62 and KDE Connect 1.33.2.
> 
> GSConnect had trouble connecting for a long time, finally got around to investigating it. I have firewall rules to only allow specific devices over IPv4, and GJS/GSConnect only listen on IPv6.
> 
> > $ netstat -lpt
> > ...
> > tcp6       0      0 [::]:1716               [::]:*                  LISTEN      7273/gjs
> 
> Would it be possible to reopen this? Or should I created a new issue?

Like @andyholmes said, GSConnect listens for both ipv4 and ipv6 -- that'll only show up in netstat as the ipv6 listen, but it'll accept connections using either protocol.

The problem you're having may be on the KDE Connect side, app version 1.33.3 is just hitting Google Play with some fixes for connection problems. Can you try updating the android side to see if that helps?

If not, please do open a new issue. But the issue isn't that GSConnect isn't accepting ipv4 connections; it is. (If you want to prove it to yourself, just `telnet 127.0.0.1 1716` on your machine running GSConnect. You'll connect successfully. (Nothing will be exchanged, just terminate the telnet session using <kbd>Ctrl</kbd>+<kbd>]</kbd>, then <kbd>q</kbd>.)

**Comment by ferdnyc** [comment]
@guillaume-uH57J9 \*nod\* F-Droid usually lags a bit behind Google Play, as I understand. But they [have the 1.33.3 update](https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/org.kde.kdeconnect_tp.yml) in their repo, so I assume it's just a matter of waiting on their build servers to get around to packaging and publishing the update.
